python/django/v1/ampli_app/ampli/ampli.py

```diff
@@ -0,0 +1,815 @@
+            "optionalNumberArray": optional_number_array,
+            "optionalStringArray": optional_string_array
+        }
+
+
+class EventWithOptionalProperties(BaseEvent):
+    """Event With Optional Properties
+
+    [View in Tracking Plan](https://data.amplitude.com/test-codegen/Test%20Codegen/events/main/latest/Event%20With%20Optional%20Properties)
+
+    Event w optional properties description
+
+    Owner: Test codegen
+
+    :param optional_array_number: Property has no description provided in tracking plan.
+    :param optional_array_string: Property has no description provided in tracking plan.
+    :param optional_boolean: Property has no description provided in tracking plan.
+    :param optional_number: Property has no description provided in tracking plan.
+    :param optional_string: Optional String property description
+    """
+
+    def __init__(
+        self,
+        optional_array_number: Optional[List[float]] = None,
+        optional_array_string: Optional[List[str]] = None,
+        optional_boolean: Optional[bool] = None,
+        optional_number: Optional[float] = None,
+        optional_string: Optional[str] = None
+    ):
+        super().__init__(event_type="Event With Optional Properties")
+        self.event_properties = {
+            "optionalArrayNumber": optional_array_number,
+            "optionalArrayString": optional_array_string,
+            "optionalBoolean": optional_boolean,
+            "optionalNumber": optional_number,
+            "optionalString": optional_string
+        }
+
+
+class Ampli:
+    
+    def __init__(self):
+        self.client: Amplitude = None
+        self.disabled: bool = False
+
+    def load(self, options: Optional[LoadOptions] = None):
+        if not options:
+            options = LoadOptions()
+        self.disabled = options.disabled
+        if self.client:
+            logging.getLogger(__name__).warning('Warning: Ampli is already initialized. ampli.load() should be called once at application start up.')
+            return
+        if not options.client:
+            options.client = LoadClientOptions(configuration=DEFAULT_CONFIGURATION)
+        api_key = None
+        if options.client.api_key:
+            api_key = options.client.api_key
+        elif options.client.instance:
+            api_key = options.client.instance.configuration.api_key
+        elif options.environment:
+            api_key = API_KEY[options.environment]
+        else:
+            api_key = API_KEY[Environment.DEVELOPMENT] or API_KEY[Environment.PRODUCTION]
+        if not api_key:
+            logging.getLogger(__name__).error("ampli.load() requires 'environment', 'client.api_key', or 'client.instance'")
+            return
+        configuration = options.client.configuration or DEFAULT_CONFIGURATION
+        self.client = options.client.instance or Amplitude(api_key=api_key, configuration=configuration)
+        if not self.client.configuration.plan:
```

Comment:
Do we need this? Should we align this on other ampli wrappers?
cc: @justin-fiedler

Reply:
Thanks @justin-fiedle, make sense. 
cc: @dnshi  let's also add the logic to ampli ts and js. I will add this in Kotlin.